tests/Client/ClientTest.php

```diff
@@ -1,62 +1,71 @@
 use Fazland\SkebbyRestClient\Constant\Endpoints;
 use Fazland\SkebbyRestClient\Constant\Recipients;
 use Fazland\SkebbyRestClient\Constant\SendMethods;
 use Fazland\SkebbyRestClient\DataStructure\Response;
 use Fazland\SkebbyRestClient\DataStructure\Sms;
+use Fazland\SkebbyRestClient\Event\SmsMessageSent;
 use Fazland\SkebbyRestClient\Exception\EmptyResponseException;
 use Fazland\SkebbyRestClient\Exception\NoRecipientsSpecifiedException;
 use Fazland\SkebbyRestClient\Exception\UnknownErrorResponseException;
 use Fazland\SkebbyRestClient\Transport\CurlExtensionTransport;
-use Kcs\FunctionMock\NamespaceProphecy;
-use Kcs\FunctionMock\PhpUnit\FunctionMockTrait;
+use Fazland\SkebbyRestClient\Transport\Psr7ClientTransport;
+use Nyholm\Psr7\Request;
+use Nyholm\Psr7\Response as Psr7Response;
+use Nyholm\Psr7\Stream;
 use PHPUnit\Framework\TestCase;
 use Prophecy\Argument;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
 use Symfony\Component\EventDispatcher\EventDispatcher;
 
-/**
- * @author Massimiliano Braglia <massimiliano.braglia@fazland.com>
- */
 class ClientTest extends TestCase
 {
-    use FunctionMockTrait;
+    use ProphecyTrait;
 
-    const RESPONSE_WITHOUT_STATUS =
-'<?xml version="1.0" encoding="UTF-8"?>
+    private const RESPONSE_WITHOUT_STATUS =
+    '<?xml version="1.0" encoding="UTF-8"?>
 <SkebbyApi_Public_Send_SmsEasy_Advanced generator="zend" version="1.0"><test_send_sms_classic_report><remaining_sms>5</remaining_sms><id>1477056680</id></test_send_sms_classic_report></SkebbyApi_Public_Send_SmsEasy_Advanced>';
 
-    const RESPONSE_FAIL =
-'<?xml version="1.0" encoding="UTF-8"?>
+    private const RESPONSE_FAIL =
+    '<?xml version="1.0" encoding="UTF-8"?>
 <SkebbyApi_Public_Send_SmsEasy_Advanced generator="zend" version="1.0"><test_send_sms_classic><response><code>11</code><message>Unknown charset, use ISO-8859-1 or UTF-8</message></response><status>failed</status></test_send_sms_classic></SkebbyApi_Public_Send_SmsEasy_Advanced>';
 
-    const RESPONSE_SUCCESS =
-'<?xml version="1.0" encoding="UTF-8"?>
+    private const RESPONSE_SUCCESS =
+    '<?xml version="1.0" encoding="UTF-8"?>
 <SkebbyApi_Public_Send_SmsEasy_Advanced generator="zend" version="1.0"><test_send_sms_classic_report><remaining_sms>5</remaining_sms><id>1477056680</id><status>success</status></test_send_sms_classic_report></SkebbyApi_Public_Send_SmsEasy_Advanced>';
 
-    /**
-     * @var array
-     */
-    private $config;
-
-    /**
-     * @var Client
-     */
-    private $skebbyRestClient;
-
-    /**
-     * @var NamespaceProphecy
-     */
-    private $functionMockNamespace;
-
-    /**
-     * {@inheritdoc}
-     */
-    protected function setUp()
+    /** @var ObjectProphecy|ClientInterface */
+    private ObjectProphecy $client;
```

Comment:
From my POV, it is preferable having it typed as `object` instead of direct `ObjectProphecy`. I know it is an instance of that class, but has it can be treated like an instance of `ClientInterface`, `object` makes it clearer

Reply:
They are `ObjectProphecy` objects explicitly. Anyway phpstan stubs have been installed to analyse these calls statically with phpstan (and comments have been changed to `ObjectProphecy<T>` instead of `ObjectProphecy|T`)